schemaless/sources.py

```diff
@@ -50,6 +55,57 @@ def get_value_str(self, record):
         return self.get_value(record).isoformat()
 
 
+class Address(Field):
+    def __init__(self, *fields):
+        self.fields = fields
+
+    def __str__(self):
+        return " ".join(self.fields)
+
+    def get_value(self, record):
+        vals = []
+        for field in self.fields:
+            if isinstance(field, Field):
+                vals.append(field.get_value_str(record))
+            else:
+                vals.append(record.get(field, ""))
+        addr_str = " ".join(vals).strip().title()
+        if not addr_str:
+            return ""
+        try:
+            addr = normalize_address_record(addr_str)
+        except UnParseableAddressError:
+            print("WARN1: Unparseable: %s" % addr_str)
+            return ""
+        else:
+            if not addr['postal_code']:
+                # We need this to normalize again, and we can't guess it
+                try:
+                    return format_address_record(addr)
+                except IncompleteAddressError:
+                    print("WARN2: Unable to format address %s" % addr)
+                    return ""
+
+            if not addr['city']:
+                addr['city'] = "San Francisco"
+            if not addr['state']:
+                addr['state'] = "California"
+
+        try:
+            addr = normalize_address_record(addr)
```

Comment:
Why do we normalize twice again?

Reply:
To make sure everything is normalized after filling in any gaps for zip/city/state right?

Reply:
There are a lot of addresses that look like "123 Main St 94102". This gets parsed into a dict that looks like:

```python
{'address_line_1': '123 MAIN ST',
 'address_line_2': None,
 'city': None,
 'state': None,
 'postal_code': '94102'}
```

We want `city` and `state` to be populated, too, so I add them in and then normalize once more to ensure it's still valid.